pySDC/helpers/fieldsIO.py

```diff
@@ -0,0 +1,731 @@
+        """
+        Parameters
+        ----------
+        dtype : np.dtype
+            The data type of the fields values.
+        fileName : str
+            File.
+        """
+        assert dtype in DTYPES_AVAIL, f"dtype not available ({dtype})"
+        self.dtype = dtype
+        self.fileName = fileName
+        self.initialized = False
+
+        # Initialized by the setHeader abstract method
+        self.header = None
+        self.nItems = None  # number of values (dof) stored into one field
+
+    def __str__(self):
+        return f"FieldsIO[{self.__class__.__name__}|{self.dtype.__name__}|file:{self.fileName}]<{hex(id(self))}>"
+
+    def __repr__(self):
+        return self.__str__()
+
+    @classmethod
+    def fromFile(cls, fileName):
+        """
+        Read a file storing fields, and return the `FieldsIO` of the appropriate
+        field type (structure).
+
+        Parameters
+        ----------
+        fileName : str
+            Name of the binary file.
+
+        Returns
+        -------
+        fieldsIO : :class:`FieldsIO`
+            The specialized `FieldsIO` adapted to the file.
+        """
+        assert os.path.isfile(fileName), f"not a file ({fileName})"
+        with open(fileName, "rb") as f:
+            STRUCT, DTYPE = np.fromfile(f, dtype=H_DTYPE, count=2)
+            fieldsIO: FieldsIO = cls.STRUCTS[STRUCT](DTYPES[DTYPE], fileName)
+            fieldsIO.readHeader(f)
+            fieldsIO.initialized = True
+        return fieldsIO
+
+    @property
+    def hBase(self) -> np.ndarray:
+        """Base header into numpy array format"""
+        return np.array([self.sID, DTYPES_AVAIL[self.dtype]], dtype=H_DTYPE)
+
+    @classmethod
+    def register(cls, sID):
+        """
+        Decorator used to register a new class FieldsIO specialized class
+
+        Parameters
+        ----------
+        sID : int
+            Unique identifyer for the file, used in the binary file.
+            Since it's encoded on a 8-bytes signed integer,
+            it must be between -128 and 127
+
+        Example
+        -------
+        >>> # New specialized FieldsIO class
+        >>> @FieldsIO.register(sID=31)
+        >>> class HexaMesh2D(FieldsIO):
+        >>>     pass # ... implementation
+        """
+
+        def wrapper(registered: Type[T]) -> Type[T]:
+            assert (
+                sID not in cls.STRUCTS
+            ), f"struct ID already taken by {cls.STRUCTS[sID]}, cannot use it for {registered}"
+            cls.STRUCTS[sID] = registered
+            registered.sID = sID
+            return registered
+
+        return wrapper
+
+    def initialize(self):
+        """Initialize the file handler : create the file with header, removing any existing file with the same name"""
+        assert self.header is not None, "header must be set before initializing FieldsIO"
+        assert not self.initialized, "FieldsIO already initialized"
+
+        with open(self.fileName, "w+b") as f:
```

Comment:
I think it would be handy to implement a check if the file already exists and add a parameter to this function to raise an error if the file already exists by default. This would protect against overwriting previous simulation results by accident.